src/app/code/community/Shopgate/Framework/Model/Shopgate/Cart/Validation/Stock/Bundle.php

```diff
@@ -49,7 +49,8 @@ public function validateStock(Mage_Sales_Model_Quote_Item $item, $priceInclTax,
             $childProduct = $childItem->getProduct();
             /** @var Mage_CatalogInventory_Model_Stock_Item $childStock */
             $childStock = $childProduct->getStockItem();
-            if ($childStock->getManageStock() && !$childProduct->isSaleable() && !$childStock->getBackorders()) {
+            if (!$childProduct->isAvailable()
```

Comment:
Wouldn't this throw a wrong error then? What does inactive have to do with being out of stock?

Reply:
if a product is inactive, it can not be sold. in check_cart and check_stock we should return out of stock.
use case: product was active and in stock, so it was exported. then it was deactivated, but check_cart returned it as valid, since it is still in stock.

also, the error is not thrown. example check_cart output

"items": [
        {
            "item_number": "18",
            "is_buyable": 0,
            "qty_buyable": 1000000,
            "stock_quantity": 1000000,
            "unit_amount": 1,
            "unit_amount_with_tax": 1,
            "error": 300,
            "error_text": "product is not in stock",
            "options": [],
            "inputs": [],
            "attributes": []
        }

Reply:
I was thinking of setting a different error when it is encountered as the use case has nothing to do with quantity, but availability.```ShopgateLibraryException::CART_ITEM_PRODUCT_NOT_FOUND``` will work better in this use case.
I am just thinking that this error will help better debugging on the support's end than saying that the item is out of stock.

Reply:
I guess that error message would be more helpful.
done
I also reduced the nesting a bit